eva_submission/submission_config.py

```diff
@@ -30,14 +37,29 @@ def set(self, *path, value):
             top_level = top_level[p]
         top_level[path[-1]] = value
 
+    def pop(self, *path, default=None):
+        """Recursive dictionary pop with default"""
+        top_level = self.content
+        for p in path[:-1]:
+            if p not in top_level:
+                return default
+            top_level = top_level[p]
+        return top_level.pop(path[-1], default)
+
     def is_empty(self):
         return not self.content
 
     def clear(self):
         self.content = {}
 
+    def __contains__(self, item):
+        return item in self.content
+
     def __setitem__(self, item, value):
         """Allow dict-style write access, e.g. config['this']='that'."""
+        # If we're starting to fill in an empty config, set the version.
```

Comment:
I'm not sure this is what we want or if we should record the latest version that was use to write the config.
Assume the following:
 - version 1 has a working `prepare submission.py` and a broken `validate_submission.py`
 - version 2 update the config so that `validate_submission.py` has a correct config

If I run the `prepare_submission.py` with version 1 and `validate_submission.py` with version 2, I won't record the correct version of the config.

Reply:
Actually, I see why this implemented as it is. We also want to avoid updating to the newest version when only a small change has been made.

Reply:
That was my thinking, so in the example you give, the config upgrade should set the version to whatever it's upgrading to, but if there are nonbreaking changes it leaves the version number alone.  Open to other options if you prefer though.

Reply:
I think it makes sense but it means that we need to be very aware of breaking changes because we will have to create upgrade scripts every time we have some.